docs/querying/caching.md

```diff
@@ -22,63 +22,75 @@ title: "Query caching"
 
-Brokers support both segment-level and whole-query result level caching. Segment-level caching is controlled by the
-parameters `useCache` and `populateCache`. Whole-query result level caching is controlled by the parameters
-`useResultLevelCache` and `populateResultLevelCache` and [runtime properties](../configuration/index.md)
-`druid.broker.cache.*`.
+When your queries include data from segments that are mutable and undergoing real-time ingestion, use a segment cache. In this case Druid caches query results for immutable historical segments when possible. It re-computes results for the real-time segments at query time.
 
-Enabling segment-level caching on the Broker can yield faster results than if query caches were enabled on Historicals for small
-clusters. This is the recommended setup for smaller production clusters (< 5 servers). Populating segment-level caches on
-the Broker is _not_ recommended for large production clusters, since when the property `druid.broker.cache.populateCache` is
-set to `true` (and query context parameter `populateCache` is _not_ set to `false`), results from Historicals are returned
-on a per segment basis, and Historicals will not be able to do any local result merging. This impairs the ability of the
-Druid cluster to scale well.
+For example, you have queries that frequently include incoming data from a Kafka or Kinesis stream alongside unchanging segments. Whole-query caching is not helpful in this scenario because the new data from real-time ingestion continually invalidates the cache. Segment caching lets Druid cache results from older immutable segments and merge them with updated data.
 
-## Query caching on Historicals
+### Whole-query caching
 
-Historicals only support segment-level caching. Segment-level caching is controlled by the query context
-parameters `useCache` and `populateCache` and [runtime properties](../configuration/index.md)
-`druid.historical.cache.*`.
+If real-time ingestion invalidating the cache is not an issue for your queries, you can use **whole-query caching** on the Broker to increase query efficiency. Segment caching is less efficient in this case because it requires Druid to merge the per-segment results for each query, even when the results are cached. For instance, whole-query caching is a good option when you have queries that include data from a batch ingestion task that runs every few hours or once a day.
 
-Larger production clusters should enable segment-level cache population on Historicals only (not on Brokers) to avoid
-having to use Brokers to merge all query results. Enabling cache population on the Historicals instead of the Brokers
-enables the Historicals to do their own local result merging and puts less strain on the Brokers.
+## Where to enable caching
 
-## Query caching on Ingestion Tasks
+**Segment cache** is available as follows:
+- On Brokers for small production clusters with less than five servers. 
```

Comment:
nit: Maybe this should be last in the list, since it's the least impactful. 

Caching on the historicals - applies to almost everyone.
Caching on the ingestion tasks - useful if you're querying realtime data
Caching on the broker - not useful in production scenarios... I can't think of a reason to ever to this tbh

Reply:
Great suggestion. especially since this moves the default to the top of the list.